python/create_service_diagnostics_bundle.sh

```diff
@@ -109,4 +109,4 @@ container_run "rm -rf ${CONTAINER_DCOS_CLI_DIRECTORY} && mkdir ${CONTAINER_DCOS_
                  -exec cp --parents \{\} ${CONTAINER_DCOS_CLI_DIRECTORY} \;
                cd /
                dcos plugin add /tmp/dcos-core-cli-${DCOS_CLUSTER_MAJOR_MINOR_VERSION}.zip
-               ${CONTAINER_SCRIPT_PATH} ${*} --bundles-directory ${CONTAINER_BUNDLES_DIRECTORY}"
+               ${CONTAINER_SCRIPT_PATH} ${*} --bundles-directory ${CONTAINER_BUNDLES_DIRECTORY} --yes"
```

Comment:
Why was `--yes` appended here?
```suggestion
               ${CONTAINER_SCRIPT_PATH} ${*} --bundles-directory ${CONTAINER_BUNDLES_DIRECTORY}"
```

Reply:
Because, actually, user will be prompted by [Y/n]? question but can't answer because only output to from container is supported now (having `-t` option).   Pls, take a look on my comment https://github.com/mesosphere/dcos-sdk-service-diagnostics/pull/12#pullrequestreview-285394527.

Reply:
Ok, I see that the comment was edited to read _the opposite_ what it was before.

In that case, I think maintaining the possibility of interactivity is more important than making a CI build pass. Maybe we could change the environment in which the script runs in CI somehow, so that the possibility of interactivity is maintained? Or fix it some other way that doesn't result in making the script always run non-interactively.

cc @porridge

Reply:
Also, to avoid confusion in the future, please add new comments instead of editing existing ones if the meaning is gonna be reverted! 😛

Reply:
Why is it important? User have been asked is he agree to install a package's dcos-cli sub-command under container. If he disagree he won't get diagnostics which he requested...
So what's a reason to ask him that?

Reply:
> Also, to avoid confusion in the future, please add new comments instead of editing existing ones if the meaning is gonna be reverted! 😛

yeah, sorry again, I was thinking (for some reason) that you did catch previous one.

Reply:
Sorry, it's not entirely clear to me what exactly that `--yes` relates to.
If it's indeed only the confirmation to install the necessary CLI plugins *inside* the throw-away container, then I agree it should have never been asked in the first place when invoked from within this shell script.

Reply:
Ah, no, AFAICT it only affects [this piece of code](https://github.com/mesosphere/dcos-sdk-service-diagnostics/blob/938bf0722c7c340028888328dc4fc65e28ab77f2/python/create_service_diagnostics_bundle.py#L223-L226) and *not* CLI plugin installation, right?

Since this is not a destructive operation, I doubt we need this confirmation at all, but let's not change the UX in the context of this ticket. Perhaps later we can ask "the field" about whether users expect this, or it's a mere annoyance.

But for now let's just pass `"$@"` to the Python script and make TeamCity pass `--yes`.

Reply:
@porridge the confirmation asking for user input before actually starting creating the bundle was an intentional feature added to the script so that the user can confirm that the details look correct. Users might have multiple clusters, or run commands with incorrect parameters.

```
$ ./create_service_diagnostics_bundle.sh --package-name=elastic --service-name=/data-services/elastic
Initializing diagnostics...
Detected attached DC/OS cluster major and minor version as '1.13'
'plugin' is not a dcos command.
Checking authentication to DC/OS cluster...
Checking attached DC/OS cluster state...
Getting scheduler information...

Will create bundle for:
  Package:         elastic
  Package version: 2.7.0-6.8.1
  Service name:    /data-services/elastic
  DC/OS version:   1.13.4
  Cluster URL:     https://soak113s.testing.mesosphe.re

Proceed? [Y/n]: 
```
The bigger point here is that existing behavior is being changed (breaking backwards compatibility) just to make a CI build pass.

Reply:
> @porridge: Ah, no, AFAICT it only affects this piece of code and not CLI plugin installation, right?

Yes.

Reply:
> @porridge: Ah, no, AFAICT it only affects this piece of code and not CLI plugin installation, right?

my bad, sorry. 

@mpereira if it was added intentionally, maybe we can move this question up and ask user from _create_service_diagnostics_bundle.sh_ before start container?

Reply:
> @VladimirKravtsov: if it was added intentionally, maybe we can move this question up and ask user from create_service_diagnostics_bundle.sh before start container?

I think a better option would make the interactive flag be an option to the script that defaults to true, and have CI pass it as false.

```
INTERACTIVE=false ./create_service_diagnostics_bundle.sh <...>
```

Do you know if it's possible to change the CI environment in some way where no changes are required and the interactive flag works? This is a similar approach to what [docker-compose](https://github.com/docker/compose/issues/5696) does.

Reply:
A similar approach is also used by [dcos-commons' test.sh script](https://github.com/mesosphere/dcos-commons/blob/ca1d2a0f09c226e06399fe4c990035473cff7875/test.sh#L309-L313).